### PR TITLE
typed blocks now auto fill default arguments

### PR DESCRIPTION
**Pull request by evanrthomas**
currently, blocks that have default arguments (like notifier.showChooseDialog) make default arguments when they are dragged form the pop-up area ,but not when they are typed in.

### CODE REVIEW

**Comment by evanrthomas** [comment]
I handled "make color" and "Lookup in pairs". Also deleted the useless comment

File: appinventor/lib/blockly/src/core/typeblock.js [hunk 1]
```diff
@@ -458,15 +459,7 @@ Blockly.TypeBlock.createAutoComplete_ = function(inputText){
...
-        if(!goog.object.isEmpty(blockToCreate.mutatorAttributes)) {
-          //construct xml
-          var xmlString = '<xml><block type="' + blockToCreateName + '"><mutation ';
-          for(var attributeName in blockToCreate.mutatorAttributes) {
-            xmlString += attributeName + '="' + blockToCreate.mutatorAttributes[attributeName] + '" ';
-          }
-
-          xmlString += '>';
-          xmlString += '</mutation></block></xml>';
+        if(xmlString = Blockly.Drawer.getDefaultXMLString(blockToCreate.canonicName, blockToCreate.mutatorAttributes)) {
```
Comment (josmas) [inline appinventor/lib/blockly/src/core/typeblock.js]: Missing `var` declaration for xmlString variable <-- leaking into window object as a global; @evanrthomas can you please check?

**Comment by evanrthomas** [comment]
@josmas I don't actually think it would leak, because this whole block of code is in one anonymous function. But it's bad practice anyway. Fixed

**Comment by josmas** [comment]
@evanrthomas it's easy to check; just see if the value exists in the window object:

leaking_var_js

**Comment by evanrthomas** [comment]
yea, it didn't end up on the window object because it was in a function. But I prefer using var anyway. I'm sure not using var is bad for some reason. Probably in some "best java script practices book" somewhere. lol. Anyway, I put the var in. Fixed

**Comment by jisqyv** [comment]
@evanrthomas @josmas LGTM, Sending to Gerrit

**Comment by jisqyv** [comment]
MERGED.

**Pull request closed without merging**